sandbox-tools/merge-repos/src/syncReposToStaging.ts

```diff
@@ -507,100 +509,222 @@ async function resolveConflictsToTheirs(git: SimpleGit, commitDetails: ICommitDe
+            let destStats = fs.statSync(destFolder + "/" + destFile);
+            if (destStats.isDirectory()) {
+                commitDetails.message = logAppendMessage(commitDetails.message, { index: "*", working_dir: "F", path: destFolder + "/" + destFile } , `Removing extra folder ${destFile}`);
+                await git.raw([
+                    "rm",
+                    "-f",
+                    "-r",
+                    destFolder + "/" + destFile]);
+                // fs.rmdirSync(destFolder + "/" + destFile, {
+                //     recursive: true
+                // });
             } else {
-                log(` - Ignoring ${inputFile}  (${destFolder})`);
+                commitDetails.message = logAppendMessage(commitDetails.message, { index: "*", working_dir: "E", path: destFolder + "/" + destFile } , "Removing extra file");
+                await git.rm(destFolder + "/" + destFile);
             }
-        };
+        }
+    }
+}
+
+async function getAndSyncSrcRepo(git: SimpleGit, repoName: string, details: IRepoDetails): Promise<ISourceRepoDetails> {
+
+    // Get a clone of the source repo and reset to the starting point
+    let srcOriginRepoUrl = details.url;
+    let srcOriginBranch = details.branch;
+
+    let forkDestOrg =  _mergeGitRoot + "-" + repoName;
+
+    // Now lets go and create a local clone
+    log(`Cloning the source repo ${srcOriginRepoUrl} branch ${srcOriginBranch} to ${forkDestOrg}`);
+    await git.clone(srcOriginRepoUrl, forkDestOrg, [ "-b", srcOriginBranch]);
+
+    let orgGit = createGit(forkDestOrg, "merge.org." + repoName + ".git");
+    let checkoutArgs = [
+        "--progress",
+        "-B", srcOriginBranch
+    ];
 
-        commitDetails.message += commitMessage;
+    if (details.branchStartPoint) {
+        // Used for testing the the consistent "merging" over time based on using the configured
+        // tags (startPoints) from the original master repo.
+        checkoutArgs.push(details.branchStartPoint);
     } else {
-        log(` - No files found in ${baseFolder + "/" + srcFolder}`);
+        checkoutArgs.push("HEAD")
     }
+
+    await orgGit.checkout(checkoutArgs);
+
+    // Reset the local branch to the requested HEAD (or mergeStartPoint -- used for testing)
+    log("Resetting...");
+    await orgGit.reset(["--hard"]).catch(abort(git, "Failed to hard reset"));
+
+    // Remove any untracked files in this local branch
+    log("Cleaning...");
+    // The excludes where for local development / branch purposes to ensure local changes where not lost
+    await orgGit.clean([CleanOptions.RECURSIVE, CleanOptions.FORCE], ["-e", "/.vs"]).catch(abort(git, "Failed during clean"));
+
+    let hashDetails = await orgGit.show(["-s", details.branchStartPoint ? details.branchStartPoint : "HEAD"]).catch(abort(git, `Failed getting hash details ${details.branchStartPoint ? details.branchStartPoint : "HEAD"}`));
```

Comment:
Grab the latest has from the local cloned repo (used for the descriptions of the PR and Commits)